### ISSUE TITLE
TraitList setitem & delitem regression in Python 3

### ISSUE THREAD

**Issue by sjagoe** [enthought/traits#334]
Following #115, there is a regression in setting a slice of `[:]` in traits lists.

Given:

```
from traits.api import HasTraits, on_trait_change, List
class ListObject(HasTraits):
    my_list = List

    @on_trait_change('my_list_items')
    def _my_list_item_changed(self, obj, old, event):
        assert event.index == 0, event.index
```

The code will fail on the assertion.

In Python 2, when the TraitList code used `__setslice__` and `__delslice__`, the event index value was set to `0` implicitly as the start value for a slice of `[:]`. The move to using only `__setitem__` and `__delitem__` changed this behaviour to use a slice of `slice(None, None, None)`.

**Comment by mdickinson** [comment]
With the other changes to `TraitListEvent`, let's review this issue for the 6.0 release.

**Comment by mdickinson** [comment]
This issue appears to be out of date. I can't reproduce on any recent version of Traits.

**Comment by mdickinson** [comment]
This was re-reported in #540, and fixed in #548. The fix went into the 5.2.0 release.
